soccer/src/soccer/strategy/agent/position/runner.cpp

```diff
@@ -0,0 +1,107 @@
+#include "runner.hpp"
+
+#include <spdlog/spdlog.h>
+
+namespace strategy {
+
+Runner::Runner(int r_id) : Position(r_id, "Runner") {}
+
+Runner::Runner(const Position& other) : Position{other} {}
+
+std::optional<RobotIntent> Runner::derived_get_task(RobotIntent intent) {
+    latest_state_ = update_state();
+    return state_to_task(intent);
+}
+
+std::string Runner::get_current_state() { return "Runner"; }
+
+bool inPosition(rj_geometry::Point current, rj_geometry::Point target, double tolerance) {
+    return current.dist_to(target) < tolerance;
```

Comment:
take a look at the `check_is_done` function. When the plan succeeds it returns true. This way thresholding for plans is abstracted away from strategy